framework/wazuh/agent.py

```diff
@@ -438,103 +460,61 @@ def get_agents_keys(agent_list: list = None) -> AffectedItemsWazuhResult:
         Result with affected agents.
     """
-    result = AffectedItemsWazuhResult(all_msg='All selected agents were deleted',
-                                      some_msg='Some agents were not deleted',
-                                      none_msg='No agents were deleted'
-                                      )
-    if agent_list:
-        system_agents = get_agents_info()
-        rbac_filters = get_rbac_filters(system_resources=system_agents, permitted_resources=agent_list,
-                                        filters=filters)
-
-        with WazuhDBQueryAgents(limit=None, select=["id"], query=q, **rbac_filters) as db_query:
-            data = db_query.run()
-
-        can_purge_agents = set(map(operator.itemgetter('id'), data['items']))
-        agent_list = set(agent_list)
-
-        try:
-            agent_list.remove('000')
-            result.add_failed_item('000', WazuhError(1703))
-        except KeyError:
-            pass
-
-        # Add not existing agents to failed_items
-        not_found_agents = agent_list - system_agents
-        list(map(lambda ag: result.add_failed_item(id_=ag, error=WazuhResourceNotFound(1701)), not_found_agents))
-
-        # Add non eligible agents to failed_items
-        non_eligible_agents = agent_list - not_found_agents - can_purge_agents
-        list(map(lambda ag: result.add_failed_item(id_=ag, error=WazuhError(
-            1731,
-            extra_message="some of the requirements are not met -> {}".format(
-                ', '.join(f"{key}: {value}" for key, value in filters.items() if key != 'rbac_ids') +
-                (f', q: {q}' if q else '')
+    result = AffectedItemsWazuhResult(
+        all_msg='All selected agents were deleted',
+        some_msg='Some agents were not deleted',
+        none_msg='No agents were deleted'
+    )
+
+    async with get_indexer_client() as indexer:
+        if len(agent_list) != 0:
+            available_agents = get_source_items_id(
+                await indexer.agents.search(query={QUERY_KEY:{TERMS_KEY:{'_id':agent_list}}})
             )
-        )), non_eligible_agents))
+            not_found_agents = set(agent_list) - set(available_agents)
 
-        for agent_id in agent_list.intersection(system_agents).intersection(can_purge_agents):
-            try:
-                my_agent = Agent(agent_id)
-                my_agent.remove(purge=purge)
-                result.affected_items.append(agent_id)
-            except WazuhError as e:
-                if e.code == 1726:
-                    raise e
+            for not_found_id in not_found_agents:
+                result.add_failed_item(not_found_id, error=WazuhResourceNotFound(1701))
 
-                result.add_failed_item(id_=agent_id, error=e)
+            agent_list = available_agents
+        else:
+            agent_list = get_source_items_id(
+                await indexer.agents.search(query={QUERY_KEY:{WILDCARD_KEY:{ID_KEY:'*'}}})
+            )
```

Comment:
Deleting all agents if no IDs were specified could be a dangerous operation, specially for users executing requests manually. Perhaps we could delete them all if the only item in the list is `all`, or adding a parameter `all`

Reply:
This function will receive an empty list from the controller when the `all` keyword is in the `agent_list` parameter.

https://github.com/wazuh/wazuh/pull/25197/files#diff-b71e39de742e4dceb17b1b4b45f7142ae712b715f7819a0965c260fdf8e1e4b4R45-R46

However, we have pending the definition of the full list of fields that we'll use to delete agents. This could be changing in the future. I noted this topic in the technical debt issue.

Reply:
I've added a clarification in the docstring for now 66ce9b92d971962c7d865db6ee0a056e6aabb523